extensions/variants.go

```diff
@@ -83,6 +86,43 @@ func EvaluateTypeExpression(nullHandling NullabilityHandling, expr parser.TypeEx
 	return outType, nil
 }
 
+func matchArguments(nullability NullabilityHandling, paramTypeList ArgumentList, actualTypes []types.Type) (bool, error) {
+	if len(paramTypeList) != len(actualTypes) {
+		return false, nil
+	}
+	for i := range paramTypeList {
+		match, err := matchArgumentAt(actualTypes[i], i, nullability, paramTypeList)
+		if err != nil {
+			return false, err
+		}
+		if !match {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+func matchArgumentAt(actualType types.Type, argPos int, nullability NullabilityHandling, paramTypeList ArgumentList) (bool, error) {
+	if argPos < 0 {
+		return false, errors.New("non-zero argument position")
+	}
+	if argPos >= len(paramTypeList) {
+		return false, fmt.Errorf("%w: argument position %d out of range", substraitgo.ErrNotFound, argPos)
```

Comment:
this should work with variadic functions. I think this failure is incorrect in that case.

Reply:
I have marked a TODO for variadic function. I need to understand behavour of variadic function more to fix

Reply:
fixed APIs to work with variadic function